lib/active_job/queue_adapters/amazon_sqs_adapter.rb

```diff
@@ -54,6 +51,25 @@ def message_attributes(job)
           }
         }
       end
+
+      def deduplication_body(job, body)
+        dedup_keys = job.deduplication_keys if job.respond_to?(:deduplication_keys)
+        dedup_keys ||= Aws::Rails::SqsActiveJob.config.deduplication_keys.map(&:to_s)
```

Comment:
I think it makes sense to use symbol throughout the method (including for job_id), and then convert to string when doing body.slice. It would eliminate to_s here and in deduplicate_with.

Reply:
Use to_s within each definition to uniquely add job_id to keys within the deduplicate_without method and excluded_deduplication_keys getter definitions.
Anyway, I agree with your ideas and would be happy to hear any other good ideas.